pimd/pim_oil.h

```diff
@@ -64,12 +64,32 @@ struct channel_counts {
 
   Each channel_oil.oil is used to control an (S,G) entry in the Kernel
   Multicast Forwarding Cache.
+
+  There is a case when we create a channel_oil but don't install in the kernel
+
+  Case where (S, G) entry not installed in the kernel:
+    FRR receives IGMP/PIM (*, G) join and RP is not configured or
+    not-reachable, then create a channel_oil for the group G with the incoming
+    interface(channel_oil.oil.mfcc_parent) as invalid i.e "MAXVIF" and populate
+    the outgoing interface where join is received. Keep this entry in the stack,
+    but don't install in the kernel(channel_oil.installed = 0).
+
+  Case where (S, G) entry installed in the kernel:
+    When RP is configured and is reachable for the group G, and receiving a
+    join if channel_oil is already present then populate the incoming interface
+    and install the entry in the kernel, if channel_oil not present, then create
+    a new_channel oil(channel_oil.installed = 1).
+
+  is_valid: indicate if this entry is valid to get installed in kernel.
+  installed: indicate if this entry is installed in the kernel.
+
 */
 
 struct channel_oil {
 	struct pim_instance *pim;
 
 	struct mfcctl oil;
+	bool is_valid;
```

Comment:
Lets consider a case: Receive IGMP/PIM join when RP is not configured or not reachable:
We will create a channel_oil with outgoing interface  where the join is received, but will not install in the kernel.

In order to populate the outgoing interface list we are calling the function pim_channel_add_oif().
But if you see pim_channel_add_oif will also try to install the mroute in the kernel by calling the function pim_mroute_add(). if fail then it will reset the outgoing interface to the old_ttl. 

In our case pim_mroute_add will fail, since there is an existing check is in pim_mroute_add()>>>> " Do not install route if incoming interface is undefined.". 
With this outgoing interface list will not be populated.

In order to avoid the calling of pim_mroute_add, i have added a is_valid check